nixos/modules/services/audio/mympd.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.services.mympd;
+in {
+  options = {
+
+    services.mympd = {
+
+      enable = lib.mkEnableOption (lib.mdDoc "MyMPD server");
+
+      package = lib.mkPackageOption pkgs "mympd" {};
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        description = lib.mdDoc ''
+          The HTTP port where mympd's web interface will be available.
+
+          The HTTPS/SSL port can be configured via {option}`config`.
+        '';
+        example = "8080";
+      };
+
+      ssl = lib.mkOption {
+        type = lib.types.bool;
+        description = lib.mdDoc ''
+          Whether to enable listening on the SSL port.
+
+          Refer to <https://jcorporation.github.io/myMPD/configuration/configuration-files#ssl-options>
+          for more information.
+        '';
+        default = false;
+      };
+
+      config = lib.mkOption {
+        type = with lib.types; attrsOf (nullOr (oneOf [ str bool int ]));
+        description = lib.mdDoc ''
+          Manages the configuration files declaratively. For all the configuration
+          options, see <https://jcorporation.github.io/myMPD/configuration/configuration-files>.
+
+          Each key represents the "File" column from the upstream configuration table, and the
+          value is the content of that file.
+
+          The configuration is automatically written at service start-up, so it is
+          possible to change their values after the first start of the service.
+        '';
+      };
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    services.mympd.config = {
+      http_port = cfg.port;
+      ssl = cfg.ssl;
+    };
+
+    systemd.services.mympd = {
+      description = "MyMPD Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = with lib; ''
+        config_dir="/var/lib/mympd/config"
+        mkdir -p "$config_dir"
```

Comment:
I think you should add a `tmpfiles` configuration for `/var/lib/mympd`. I was about to almost suggest to make the path to this option configurable, but I guess it is not possible if we insist upon using `StateDirectory` and `CacheDirectory` Systemd options... This is the conclusion I reached after reading this:

https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RuntimeDirectory=

Reply:
In the first version of this service, I actually used the classic method of `user`, `group` which did make it possible to configure these two paths; however I ended up using `DynamicUser` because it feels leaner.

I didn't add `systemd.tmpfiles` since we are using a dynamic user, so I don't know what values to provide to `user` and `group`.

Reply:
> In the first version of this service, I actually used the classic method of `user`, `group` which did make it possible to configure these two paths; however I ended up using `DynamicUser` because it feels leaner.
> 
> I didn't add `systemd.tmpfiles` since we are using a dynamic user, so I don't know what values to provide to `user` and `group`.

Hmm maybe we should use non dynamic users? That would make it easier to setup the tmpfiles directory, and it would also allow for users of this service to use a user and a group as they wish. I read that upstream sort of recommends to use a `DynamicUser`, but I don't think there will be problems in using a non dynamic user.

I'm almost sure it should also be possible to use this way a path other than `/var/lib/mympd`...

Reply:
@pbsds , what's your take on this? Should we use DynamicUser or user-configurable user/group?
I could do either of them, IMHO since i already had it implemented.